src/common/raft/integration/raft_knowhere_index.cuh

```diff
@@ -423,12 +445,22 @@ struct raft_knowhere_index<IndexKind>::impl {
 
         auto device_bitset =
             std::optional<raft::core::bitset<knowhere_bitset_data_type, knowhere_bitset_indexing_type>>{};
+        auto k_tmp = k;
 
         if (bitset_data != nullptr && bitset_byte_size != 0) {
             device_bitset =
                 raft::core::bitset<knowhere_bitset_data_type, knowhere_bitset_indexing_type>(res, bitset_size);
             raft::copy(res, device_bitset->to_mdspan(), raft::make_host_vector_view(bitset_data, bitset_byte_size));
-            device_bitset->flip(res);
+            if constexpr (index_kind == raft_proto::raft_index_kind::brute_force) {
+                k_tmp += device_bitset->count(res);
```

Comment:
It's truly a brute force approach.🤣

Reply:
Haha, yeah. If this is overkill, we can cap the maximum k at some reasonable value. This was just a way to make very, very sure that brute force returns an exact result via postfiltering until we get prefiltering in.